### DISCUSSION TITLE
Snap Sync

### DISCUSSION THREAD

**Discussion by andreclaro** [celo-org/celo-blockchain#2313] [Help (Q&A)] [answered]
### Context

Spin-up an archive node (`--gcmode archive`) with snap sync (`--syncmode snap`)

### Problem

We just want to keep the last month worth of blocks.

### Asks

Does Celo support snap sync?

**Comment by piersy** [comment, marked as answer]
Hi @andreclaro

> We just want to keep the last month worth of blocks

This is not possible with celo or even vanilla go-ethereum and I'm not sure it's possible with any ethereum client. As far as I know all clients keep all blocks.

What you may be referring to is the state for each block which if not running with `--gcmode archive` will expire after 128 blocks so for celo that would be about 10 minutes, or 25 minutes for ethereum because block times are longer.  If running with `--gcmode archive` then state never expires, unfortunately there is no extra configurability. So either you keep the last 128 sates or you keep them all.

> Does Celo support snap sync?

The celo-blockchain celo node implementation does not currently support snap sync. We are currently in the process of transitioning to be an L2 built on the Optimism stack, that new implementation will support snap sync but it is not ready yet. Although snap sync will not enable you to keep just the last months worth of blocks.

**Reply by andreclaro** [reply]
Hello @piersy , thanks!

When will Celo transition to L2? Are you going to use the original Op-Stack or fork it like other L2 chains (mantle, blast, etc.)?

Cheers
